Fix PlayerMPD.prev/next() when stopped

hoffie opened the pull request:
Fix PlayerMPD.prev/next() when stopped

* Avoid MPD-related crashes during all prev/next() calls.
* Explicitly handle prev() in stopped state, configurable via 
  `playermpd.stopped_prev_action`.
* Explicitly handle next() in stopped state, configurable via 
  `playermpd.stopped_next_action`.
* Explicitly handle next() when reaching the end of the playlist:
  jukebox-daemon will now ignore the action by default (similar to v2).
  It can also be configured to rewind the playlist instead by setting
  the new config option `playermpd.end_of_playlist_next_action: rewind`
  or to stop playing.

Fixes #2294
Fixes #2327

pabera commented on src/jukebox/components/playermpd/__init__.py:
```diff
@@ -231,6 +236,14 @@ def decode_2nd_swipe_option(self):
+        if end_of_playlist_next_action in self.end_of_playlist_next_action_dict:
+            self.end_of_playlist_next_action = self.end_of_playlist_next_action_dict[end_of_playlist_next_action]
+        else:
+            logger.error(f'Config mpd.end_of_playlist_next_action must be one of '
+                         f'{self.end_of_playlist_next_action.keys()}. Ignore setting.')
```
```suggestion
            self.end_of_playlist_next_action = self.end_of_playlist_next_action_dict['stop']
            logger.error(f"Config mpd.end_of_playlist_next_action must be one of "
                         f"{self.end_of_playlist_next_action.keys()}. "
                         f"Using default 'stop'")
```

pabera replied:
Maybe we can be a little bit more specific here. Thoughts?

hoffie replied:
I've committed your suggestion.

pabera reviewed (approved):
If there is no additional work required on this PR, this can be merged.

s-martin commented:
No action could be a third option, but that could also be added in a later PR.

https://github.com/MiczFlor/RPi-Jukebox-RFID/issues/2327#issuecomment-2050522393

hoffie commented:
> No action could be a third option, but that could also be added in a later PR.

Added now.

As there have been further changes for #2327, this PR needs another review and new testing (I haven't tested the most recent state yet, maybe @igorwalton can help?).

pabera committed:
utils: Add get_config_action

This abstracts away the functionality to resolve a given config option
to an action in a pre-defined dict.

Co-authored-by: Christian Hoffmann <christian@hoffie.info>

hoffie committed:
Fix PlayerMPD.prev/next() when stopped

* Avoid MPD-related crashes during all prev/next() calls.
* Explicitly handle prev() in stopped state, configurable via
  `playermpd.stopped_prev_action`.
* Explicitly handle next() in stopped state, configurable via
  `playermpd.stopped_next_action`.
* Explicitly handle next() when reaching the end of the playlist:
  jukebox-daemon will now ignore the action by default (similar to v2).
  It can also be configured to rewind the playlist instead by setting
  the new config option `playermpd.end_of_playlist_next_action: rewind`
  or to stop playing.

Fixes #2294
Fixes #2327

Co-authored-by: pabera <1260686+pabera@users.noreply.github.com>

hoffie commented on src/jukebox/components/playermpd/__init__.py:
```diff
@@ -231,6 +245,31 @@ def decode_2nd_swipe_option(self):
+                         f"{self.stopped_next_action.keys()}. "
+                         f"Using default 'next'")
+        self.stopped_next_action = self.stopped_prev_action_dict[stopped_prev_action]
+
```
Nice idea! I've taken your standalone function and put it into jukebox.utils, where similar functions live. I've reworked the prev/next logic to make use of that function directly.
